app/services/ImportServiceImpl.java

```diff
@@ -0,0 +1,142 @@
+import play.libs.F;
+import utils.JsonUtils;
+
+import javax.inject.Inject;
+import java.util.Optional;
+
+import static java.util.Objects.requireNonNull;
+
+@Singleton
+public class ImportServiceImpl extends AbstractShopService implements ImportService {
+
+    private static final Logger.ALogger LOG = Logger.of(ImportServiceImpl.class);
+
+    private static final String PRODUCT_TYPE_JSON_RESOURCE = "data/product-type-draft.json";
+    private static final String TAX_CATEGORY_JSON_RESOURCE = "data/tax-category-draft.json";
+    private static final String PRODUCT_JSON_RESOURCE = "data/product-draft.json";
+    private static final String TYPE_DRAFT_JSON_RESOURCE = "data/type-draft.json";
+
+    private static final String CUSTOM_TYPE_KEY = "cart-frequency-key";
+    private static final long ALLOWED_TIMEOUT = 5000;
+
+    @Inject
+    public ImportServiceImpl(final PlayJavaSphereClient playJavaSphereClient, final Configuration configuration) {
+        super(playJavaSphereClient);
+        requireNonNull(configuration);
+        final Boolean importEnabled = configuration.getBoolean("fixtures.import.enabled", false);
+        LOG.debug("Import enabled: {}", importEnabled);
+        if(importEnabled) {
+            importData();
+            importCustomType();
+        }
+    }
+
+    private void importData() {
+        LOG.debug("Starting Product import");
+        productExists().flatMap(productExits -> {
+            LOG.debug("Existing Product found: {}", productExits);
+            return (productExits) ?
+                    F.Promise.pure(null) :
+                    exportProductModel().map(product -> {
+                        LOG.debug("Finished Product import, created '{}'",
+                                product.toProjection(ProductProjectionType.CURRENT).getName());
+                        return null;
+                    });
+        }).get(ALLOWED_TIMEOUT);
+    }
+
+    private void importCustomType() {
+        LOG.debug("Starting custom Type import");
+        customTypeExists().flatMap(customTypeExists -> {
+            LOG.debug("Existing custom Type found: {}", customTypeExists);
+            return (customTypeExists) ?
+                    F.Promise.pure(null) :
+                    exportCustomType().map(type -> {
+                        LOG.debug("Finished custom Type import, created '{}'", type.getKey());
+                        return null;
+                    });
+        }).get(ALLOWED_TIMEOUT);
+    }
+
+    private F.Promise<Boolean> customTypeExists() {
+        final TypeQuery query = TypeQuery.of();
+        final F.Promise<PagedQueryResult<Type>> execute = playJavaSphereClient().execute(query);
+        return execute.map(typePagedQueryResult -> {
+            final Optional<Type> optionalType = typePagedQueryResult.getResults().stream()
+                    .filter(type -> CUSTOM_TYPE_KEY.equals(type.getKey())).findFirst();
+            return optionalType.isPresent();
+        });
+    }
```

Comment:
I would rather return the promise of the custom type instead of losing it. This way it's up to the method calling the import to decide what to do with it. Also it makes the code easier to understand IMO.

```
    private F.Promise<Type> importCustomType() {
        LOG.debug("Starting custom Type import");
        return currentCustomType().flatMap(currentCustomType -> {
            LOG.debug("Existing custom Type found: {}", currentCustomType.isPresent());
            return currentCustomType
                    .map(type -> F.Promise.pure(currentCustomType.get()))
                    .orElseGet(() -> {
                        final F.Promise<Type> typePromise = exportCustomType();
                        typePromise.onRedeem(type -> LOG.debug("Finished custom Type import, created '{}'", type.getKey()));
                        return typePromise;
                    });
        });
    }

    private F.Promise<Optional<Type>> currentCustomType() {
        final TypeQuery query = TypeQuery.of();
        final F.Promise<PagedQueryResult<Type>> execute = playJavaSphereClient().execute(query);
        return execute.map(typePagedQueryResult ->
                typePagedQueryResult.getResults().stream()
                    .filter(type -> CUSTOM_TYPE_KEY.equals(type.getKey()))
                    .findFirst()
        );
    }
```